src/TwigComponent/src/EventListener/DataModelPropsSubscriber.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+use Symfony\UX\TwigComponent\ComponentStack;
+use Symfony\UX\TwigComponent\Event\PreMountEvent;
+use Symfony\UX\TwigComponent\Util\ModelBindingParser;
+
+/**
+ * Parses the "data-model" key, which triggers extra props to be passed in.
+ *
+ * For example, data-model: "value:content" would cause a new "value" prop
+ * to be passed in set to the parent component's "content" prop.
+ *
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ *
+ * @experimental
+ *
+ * @internal
+ */
+final class DataModelPropsSubscriber implements EventSubscriberInterface
```

Comment:
Curious why you chose to put this in twig components? Is there value in having it here or just to make the transition from twig to live easier/consistent?